src/cascadia/TerminalApp/PowershellCoreProfileGenerator.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#pragma once
+#include "PowershellCoreProfileGenerator.h"
+#include "LegacyProfileGeneratorNamespaces.h"
+#include "../../types/inc/utils.hpp"
+#include "../../inc/DefaultSettings.h"
+#include "Utils.h"
+#include "DefaultProfileUtils.h"
+
+using namespace ::TerminalApp;
+
+// Legacy GUIDs:
+//   - PowerShell Core       574e775e-4f2a-5b96-ac1e-a2962a402336
+
+std::wstring_view PowershellCoreProfileGenerator::GetNamespace()
+{
+    return PowershellCoreGeneratorNamespace;
+}
+
+// Method Description:
+// - Checks if pwsh is installed, and if it is, creates a profile to launch it.
+// Arguments:
+// - <none>
+// Return Value:
+// - a vector with the PowerShell Core profile, if available.
+std::vector<TerminalApp::Profile> PowershellCoreProfileGenerator::GenerateProfiles()
+{
+    std::vector<TerminalApp::Profile> profiles{};
+
+    std::filesystem::path psCoreCmdline{};
+    if (_isPowerShellCoreInstalled(psCoreCmdline))
+    {
+        auto pwshProfile{ CreateDefaultProfile(L"PowerShell Core") };
+
+        pwshProfile.SetCommandline(std::move(psCoreCmdline));
+        pwshProfile.SetStartingDirectory(DEFAULT_STARTING_DIRECTORY);
+        pwshProfile.SetColorScheme({ L"Campbell" });
+
+        // If powershell core is installed, we'll use that as the default.
+        // Otherwise, we'll use normal Windows Powershell as the default.
+        profiles.emplace_back(pwshProfile);
+    }
+
+    return profiles;
+}
+
+// Function Description:
+// - Returns true if the user has installed PowerShell Core. This will check
+//   both %ProgramFiles% and %ProgramFiles(x86)%, and will return true if
+//   powershell core was installed in either location.
+// Arguments:
+// - A ref of a path that receives the result of PowerShell Core pwsh.exe full path.
+// Return Value:
+// - true iff powershell core (pwsh.exe) is present.
+bool PowershellCoreProfileGenerator::_isPowerShellCoreInstalled(std::filesystem::path& cmdline)

```

Comment:
While this works to detect pwsh when it's installed via MSI (either directly or through chocolatey), it doesn't pick up on it when it has been installed by scoop ( see https://scoop.sh/ ) -- this is because it's installed in a portable manner. I think scoop is quite popular, but perhaps under the radar for some.

Reply:
I think we have a followup issue for that... we've been blocked on it because we never had these dynamic generators. :smile:

Personally, I'm very excited for it automatically adding a new entry for every powershell core version installed, and perhaps even calling out its method of installation:

```
PowerShell Core 6.2.2
PowerShell 7.0
PowerShell Preview 7 (Scoop)
```

(Yeah: I really want us to follow their branding and change from "Core" to "" as of 7.0, when they are also doing so.)

Reply:
@carlos-zamora for that there's two options:
1. Create a DPG for each powershell core version. The user can then disable the ones you don't want with `disabledProfileSources`
2. Have one singular DPG for all pscore versions. The user can then add `"hidden":true` to each of the profiles that you don't want to have appear.